packages/frontend/src/components/Layout/DemoPageModal.tsx

```diff
@@ -0,0 +1,95 @@
+import { useCallback, useEffect, useState } from 'react'
+import { BiBulb } from 'react-icons/bi'
+import { useNavigate } from 'react-router-dom'
+import { useMutation } from '@apollo/client'
+import {
+  Flex,
+  Image,
+  Modal,
+  ModalContent,
+  ModalFooter,
+  ModalOverlay,
+  Text,
+} from '@chakra-ui/react'
+import { Badge, BadgeLeftIcon, Button } from '@opengovsg/design-system-react'
+import demoModalImg from 'assets/demo-modal.png'
+import * as URLS from 'config/urls'
+import { CREATE_FLOW_TEMPLATE } from 'graphql/mutations/create-flow-template'
+import {
+  FLOW_TEMPLATES_MAP,
+  FORMSG_POSTMAN_TEMPLATE,
+} from 'helpers/flow-templates'
+
+interface DemoPageModalProps {
+  onClose: () => void
+}
+
+// only 1 default template and demo video to display for GGWP v1
+const { flowName, trigger, actions, demoVideoId } =
+  FLOW_TEMPLATES_MAP[FORMSG_POSTMAN_TEMPLATE]
+
+export default function DemoPageModal(props: DemoPageModalProps) {
+  const { onClose } = props
+  const navigate = useNavigate()
+  const [flowId, setFlowId] = useState('')
+
+  const [createFlowTemplate, { loading }] = useMutation(CREATE_FLOW_TEMPLATE)
+  const onCreateFlowTemplate = useCallback(async () => {
+    const response = await createFlowTemplate({
+      variables: {
+        input: {
+          flowName,
+          trigger,
+          actions,
+          demoVideoId,
+        },
+      },
+    })
+    setFlowId(response.data?.createFlowTemplate?.id)
```

Comment:
why can't u just call navigate here?

Reply:
i think i was testing some logic and yes i can do it here thks